ykrt/src/location.rs

```diff
@@ -273,6 +273,64 @@ impl Location {
             }
         }
     }
+
+    /// If the location is in the `Compiling` state, then wait for compilation to complete.
+    ///
+    /// This is used in testing, where we need some way to tame non-determinism in order to write
+    /// meaningful tests.
+    ///
+    /// FIXME: Note that we can't simply wait for the location to move into a non-`Compiling`
+    /// state, as the compilation thread is (currently) unable to update the `HotLocation` on it's
+    /// own accord: currently the system relies on an interpreter thread to notice that compilation
+    /// is done and move the location into the next state.
+    #[cfg(feature = "c_testing")]
+    pub fn block_on_pending_compilation(&self) {
```

Comment:
I think this might be better called `block_if_compiling`?

Reply:
huh, that's odd. I renamed that to `block_on_compilation()` at some point. I must have accidentally undone it.

`block_if_compiling()` is also fine by me, so let's use that.

Reply:
Fixed in cbb5403a34